### PR TITLE
Titus tasks for gerolstein

### PR DESCRIPTION
**Pull request by TitusKirch**
N/A

### CODE REVIEW

**Review by simonostendorf — APPROVED** [review]
lgtm

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -35,16 +35,16 @@ public function index(): Response
...
-        if (Auth::user()->is_tutor) {
-            $neededPassword = config('app.tutor_password');
+        $neededPassword = '';
 
-            // check if user is admin
-            if (Auth::user()->is_admin) {
-                $neededPassword = config('app.admin_password');
-            }
+        // check we need a password
+        if (Auth::user()->hasRole(['admin'])) {
```
Comment (simonostendorf) [inline app/Http/Controllers/DashboardController.php:41]: Should super admin also have a password?

File: app/Http/Middleware/IsLoggedInAdmin.php [hunk 2]
```diff
@@ -17,12 +17,12 @@ public function handle(Request $request, Closure $next): Response
         $user = $request->user();
 
         // check if user is no tutor or is not logged in as tutor
-        if (! $user->is_tutor || ! $request->session()->has('tutor')) {
+        if (! $user->hasRole(['admin', 'esa', 'stage tutor', 'tutor']) || ! $request->session()->has('tutor')) {
```
Comment (simonostendorf) [inline app/Http/Middleware/IsLoggedInAdmin.php:20]: Should super admin be listed here?

File: app/Http/Middleware/IsLoggedInTutor.php [hunk 3]
```diff
@@ -17,7 +17,7 @@ public function handle(Request $request, Closure $next): Response
         $user = $request->user();
 
         // check if user is no tutor or is not logged in as tutor
-        if (! $user->is_tutor || ! $request->session()->has('tutor')) {
+        if (! $user->hasRole(['admin', 'esa', 'stage tutor', 'tutor']) || ! $request->session()->has('tutor')) {
```
Comment (simonostendorf) [inline app/Http/Middleware/IsLoggedInTutor.php:20]: see above

File: app/Http/Middleware/RedirectIfTutor.php [hunk 4]
```diff
@@ -17,7 +17,7 @@ public function handle(Request $request, Closure $next): Response
         $user = $request->user();
 
         // check if user is tutor and is logged in as tutor
-        if ($user->is_tutor && $request->session()->has('tutor')) {
+        if ($user->hasRole(['admin', 'esa', 'stage tutor', 'tutor']) && $request->session()->has('tutor')) {
```
Comment (simonostendorf) [inline app/Http/Middleware/RedirectIfTutor.php:20]: see above

File: resources/js/pages/Dashboard/Index.vue [hunk 5]
```diff
@@ -6,7 +6,13 @@
       hier findest du eine Übersicht über alle Veranstaltungen.
     </template>
 
-    <template v-if="user.is_tutor">
+    <template
+      v-if="
+        user.rolesArray.some((role) =>
+          ['admin', 'esa', 'stage tutor', 'tutor'].includes(role)
```
Comment (simonostendorf) [inline resources/js/pages/Dashboard/Index.vue:12]: See above

**Review by simonostendorf — COMMENTED** [review]
If you want to add role `admin` and `super admin` to a super admin user my comments are irrelevant :)

File: app/Http/Middleware/IsLoggedInAdmin.php (hunk 2 above)
Comment (TitusKirch) [reply app/Http/Middleware/IsLoggedInAdmin.php:20]: no

File: app/Http/Controllers/DashboardController.php (hunk 1 above)
Comment (TitusKirch) [reply app/Http/Controllers/DashboardController.php:41]: nope

**Comment by TitusKirch** [comment]
> If you want to add role `admin` and `super admin` to a super admin user my comments are irrelevant :)

This is exactly what we do, super admin was currently only used to prevent removal.

**Review by simonostendorf — APPROVED** [review]

**Review by simonostendorf — APPROVED** [review]

**Review by simonostendorf — APPROVED** [review]

**Pull request merged**